File: CPP version/Item.cpp
#include "Item.h"


Item::Item()
{
	name.assign("Base item");
	InstType = BaseItem;
}

Item::Item(std::string name)
{
	this->name.assign(name);
	InstType = BaseItem;

}

Item::~Item()
{
	std::cout << "base item destroyed" << std::endl;
}


void Item::PrintEffect()
{
	std::cout << "This item has no effect" << std::endl;
}

int Item::GetType()
{
	return InstType;
}



int Item::GetEffect()
{
	return 0;
}

File: Java version/Item.java

enum ItemType {
    HealingItem,
    DmgIncreaseItem,
    MaxHPIncreaseItem
}


public abstract class Item {

    protected String Name;
    ItemType InstType;

    Item(ItemType InstType, String Name) {
        this.Name = Name;
        this.InstType = InstType;
    }

    //text output methods
    public abstract void PrintEffect();

    //gameplay methods
    public abstract int GetEffect();
    public abstract ItemType GetType();

}
